### PR TITLE
Updating go.mods in release/1.18.x after modules have been released.

### PR DESCRIPTION
**Pull request by jmurret**
### Description

The following modules have been released:
- [api/v1.28.4](https://github.com/hashicorp/consul/releases/tag/api%2Fv1.28.4)
- [envoyextensions/v0.6.2](https://github.com/hashicorp/consul/releases/tag/envoyextensions%2Fv0.6.2)
- [troubleshoot/v0.6.3](https://github.com/hashicorp/consul/releases/tag/troubleshoot%2Fv0.6.3)

This PR has the updated go.mods in preparation for cutting the 1.18.14+ent release branch.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -27,7 +27,7 @@ require (
 	github.com/coreos/go-oidc/v3 v3.9.0
 	github.com/deckarep/golang-set/v2 v2.3.1
 	github.com/docker/go-connections v0.4.0
-	github.com/envoyproxy/go-control-plane v0.11.1
+	github.com/envoyproxy/go-control-plane v0.12.0
```
Comment (zalimeni) [inline go.mod:30]: Probably fine but notable. There’s a bunch of changes in https://github.com/envoyproxy/go-control-plane/releases/tag/v0.12.0, sadly they don’t seem to differentiate them now in the changelog, but nothing explicitly flagged as breaking.

File: api/go.mod [hunk 2]
```diff
@@ -11,8 +11,8 @@ retract v1.28.0 // tag was mutated
 
 require (
 	github.com/google/go-cmp v0.5.9
-	github.com/hashicorp/consul/proto-public v0.5.1
-	github.com/hashicorp/consul/sdk v0.15.0
+	github.com/hashicorp/consul/proto-public v0.6.1
+	github.com/hashicorp/consul/sdk v0.16.0
```
Comment (ndhanushkodi) [inline api/go.mod:15]: sdk looks like its at 0.16.0 and 0.16.1 in different parts of this PR, would we want to standardize on one?

Comment (jmurret) [reply api/go.mod:15]: good call.  1.16.0 is the correct max which means the submodules are correct but these root and test* go.mods should be 1.16.0

Comment (jmurret) [reply api/go.mod:15]: this is fixed in e9d89f205be9ede7fb9e37207af41b0199b98606

thank you for calling this out.

File: go.mod (hunk 1 above)
Comment (jmurret) [reply go.mod]: I reverted this in 2e06b92c43a11fa6419434a15fb4f519afb3cff4.  I'm not sure when this changed during the process.  Thank you for calling it out.

**Review by ndhanushkodi — APPROVED** [review]

**Pull request merged**